### PR TITLE
Refactor CLI to use `npm` and `yarn` instead of third party packages

### PR DESCRIPTION
**Pull request by felipedeboni**
Issue: #3274 and #3060

## What I did
1. ```yarn remove latest-version``` on the ```lib/cli``` folder
2. Created a new file ```lib/cli/lib/latest_version.js``` and used the ```has_yarn.js``` to check if yarn is actually available.
3. ```yarn test``` on root folder.

## How to test

Is this testable with jest or storyshots?
I didn't touched on tests, seems that tests are already written.

Does this need a new example in the kitchen sink apps?
No.

Does this need an update to the documentation?
No.

@danielduan you're the man to contact, so can you review if everything is fine here? If not, please let me know and I am going to fix it.

### CODE REVIEW

**Review by danielduan — APPROVED** [review]

**Comment by danielduan** [comment]
@Hypnosphi this is the only obvious thing that I think could cause the linked issue. You worked with the CLI quite a bit. Does this look good?

**Comment by Hypnosphi** [comment]
The passing CLI tests are an extremely good sign

File: lib/cli/lib/latest_version.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { sync as spawnSync } from 'cross-spawn';
+import hasYarn from './has_yarn';
+
+const packageManager = hasYarn() ? 'yarn' : 'npm';
+
+export default async function latestVersion(packageName) {
+  const result = spawnSync(packageManager, ['info', packageName, '--json'], {
```
Comment (Hypnosphi) [inline lib/cli/lib/latest_version.js]: is there any reason not to use `${packageManager} info ${packageName} version`?

**Comment by felipedeboni** [comment]
@Hypnosphi it's just to be safe. Because I read 
https ://github.com/yarnpkg/yarn/issues/4044, and it's under investigation

**Comment by Hypnosphi** [comment]
@felipedeboni output without `version` is different in the same way:

```
$ yarn info @storybook/react --json
{"type":"inspect","data":{...

$ npm info @storybook/react --json
{
  "name": "@storybook/react",...
```

**Comment by felipedeboni** [comment]
@Hypnosphi, you're right. Going to open another pull request, to actually fix NPM and also change that behavior. 

Do you want to check the code first? [Commit](https://github.com/felipedeboni/storybook/commit/c6e40d8517c22614d21d6d4564976d29dbaa666b)

**Comment by Hypnosphi** [comment]
https://github.com/storybooks/storybook/pull/3297 should already have fixed this

**Pull request merged**